exchange-v2/contracts/ExchangeBulkV2.sol

```diff
@@ -0,0 +1,215 @@
+        INftTransferProxy _transferProxy,
+        IERC20TransferProxy _erc20TransferProxy,
+        IWyvernExchange  _wyvernExchange,
+        uint newProtocolFee,
+        address newDefaultFeeReceiver,
+        IRoyaltiesProvider newRoyaltiesProvider
+    ) external initializer {
+        __Context_init_unchained();
+        __Ownable_init_unchained();
+        __TransferExecutor_init_unchained(_transferProxy, _erc20TransferProxy);
+        __RaribleTransferManager_init_unchained(newProtocolFee, newDefaultFeeReceiver, newRoyaltiesProvider);
+        __OrderValidator_init_unchained();
+        wyvernExchange = _wyvernExchange;
+    }
+
+    /*Transfer by ExchangeV2 array orders */
+    function matchOrdersBulk(
+        LibOrder.Order[] memory sellOrders,
+        bytes[] memory sellOrderSignatures,
+        address buyer
+    ) external payable {
+        uint totalAmount;
+        for (uint256 i = 0; i < sellOrders.length; i++) {
+            validateFull(sellOrders[i], sellOrderSignatures[i]);
+            LibOrder.Order memory buyerOrder;
+            buyerOrder.maker = buyer;
+            buyerOrder.makeAsset = sellOrders[i].takeAsset;
+            buyerOrder.takeAsset = sellOrders[i].makeAsset;
+            buyerOrder.dataType = 0xffffffff;
+            totalAmount += matchAndTransferBulk(sellOrders[i], buyerOrder);
+        }
+        if (msg.value > totalAmount) {
+            address(_msgSender()).transferEth(msg.value.sub(totalAmount));
+        }
+    }
+
+    /*Buy order is generate inside*/
+    function matchWyvernExchangeBulk(WyvernOrder[] memory orders) external payable {
+        uint ethAmount = msg.value;
+        for(uint i = 0; i < orders.length; i ++) {
+            matchWyvernExchange(orders[i], ethAmount);
+            ethAmount = address(this).balance;
+        }
+        if (ethAmount > 0) {
+            _msgSender().transfer(ethAmount);
+        }
+    }
+
+    /*Buy order is generate outside*/
+    function matchWyvernExchangeBulk2(WyvernOrders[] memory orders) external payable {
+        uint ethAmount = msg.value;
+        for(uint i = 0; i < orders.length; i ++) {
+            wyvernExchange.atomicMatch_{ value: ethAmount }(
+                orders[i].addrs,
+                orders[i].uints,
+                orders[i].feeMethodsSidesKindsHowToCalls,
+                orders[i].calldataBuy,
+                orders[i].calldataSell,
+                orders[i].replacementPatternBuy,
+                orders[i].replacementPatternSell,
+                orders[i].staticExtradataBuy,
+                orders[i].staticExtradataSell,
+                orders[i].vs,
+                orders[i].rssMetadata);
+            ethAmount = address(this).balance;
+        }
+        if (ethAmount > 0) {
+            _msgSender().transfer(ethAmount);
+        }
+    }
+
+    /*
+    */
+    function matchWyvernExchange(
+        WyvernOrder memory order,
+        uint ethAmount
+    ) internal {
+        WyvernOrders memory wyvernOrders;
```

Comment:
why do we need this variable? can't we form the data at the time of calling atomicMatch_() ?

Reply:
done, I don`t this variable now